test/Sema/struct_equatable_hashable.swift

```diff
@@ -0,0 +1,144 @@
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: cp %s %t/main.swift
+// RUN: %target-swift-frontend -typecheck -verify -primary-file %t/main.swift %S/Inputs/struct_equatable_hashable_other.swift -verify-ignore-unknown
+
+struct Point: Hashable {
+  let x: Int
+  let y: Int
+}
+
+if Point(x: 1, y: 2) == Point(x: 2, y: 1) { }
+var pointHash: Int = Point(x: 3, y: 5).hashValue
+
+struct Pair<T: Hashable>: Hashable {
+  let first: T
+  let second: T
+
+  func same() -> Bool {
+    return first == second
+  }
+}
+
+let p1 = Pair(first: "a", second: "b")
+let p2 = Pair(first: "a", second: "c")
+if p1 == p2 { }
+var pairHash: Int = p1.hashValue
+
+func localStruct() -> Bool {
+  struct Local: Equatable {
+    let v: Int
+  }
+
+  return Local(v: 5) == Local(v: 4)
+}
+
+struct CustomHashable: Hashable {
+  let x: Int
+  let y: Int
+
+  var hashValue: Int { return 0 }
+
+  static func ==(x: CustomHashable, y: CustomHashable) -> Bool { return true }
+}
+
+if CustomHashable(x: 1, y: 2) == CustomHashable(x: 2, y: 3) { }
+var custHash: Int = CustomHashable(x: 1, y: 2).hashValue
+
+// Check use of an struct's synthesized members before the struct is actually declared.
+struct UseStructBeforeDeclaration {
+  let eqValue = StructToUseBeforeDeclaration(v: 4) == StructToUseBeforeDeclaration(v: 5)
+  let hashValue = StructToUseBeforeDeclaration(v: 1).hashValue
+}
+struct StructToUseBeforeDeclaration: Hashable {
+  let v: Int
+}
+
+// Check structs from another file in the same module.
+if FromOtherFile(v: "a") == FromOtherFile(v: "b") {}
+let _: Int = FromOtherFile(v: "c").hashValue
+
+func getFromOtherFile() -> AlsoFromOtherFile { return AlsoFromOtherFile(v: 4) }
+if AlsoFromOtherFile(v: 3) == getFromOtherFile() {}
+
+// FIXME: This should work.
+func overloadFromOtherFile() -> YetAnotherFromOtherFile { return YetAnotherFromOtherFile(v: 1.2) }
+func overloadFromOtherFile() -> Bool { return false }
```

Comment:
Oops, forgot to go back and remove that after pulling the examples over from the `enum` version of the tests. (I'm guessing the bug that was the reason for the `FIXME` in the `enum` version was fixed but the comment mistakenly remained in that file?)

I've removed the `FIXME` from the other file as well.